src/main/java/dev/jlibra/admissioncontrol/transaction/Transaction.java

```diff
@@ -0,0 +1,67 @@
+package dev.jlibra.admissioncontrol.transaction;
+
+public class Transaction {
```

Comment:
What do you think about using [Lombok](https://projectlombok.org) to reduce boilerplate getters/withers?

Reply:
We discussed with @ice09 this briefly and decided not to include lombok then. I would not want to add too many transitive dependencies for other projects that might use jlibra. If lombok can be added so that it's required only in jlibra compilation and not as a dependency, I might consider it.

It also adds some development overhead because support for lombok needs to be installed in the ide for it to compile. For me it's been pretty easy in eclipse but I've witnessed some devs with IntelliJ struggle to get it to work. 

But this is also related to the value object pattern that we should choose so we can discuss this further in the issue I will create about that.

Reply:
Also thought about Lombok and really like it, but I agree with @ketola that we should try to keep dependencies at a minimum as this is on library level. It's possible to prevent transitive dependencies with Lombok, but it's a little hacky with _provided_ so I'd go for resisting here and introducing it on a "user level" rather than on core/library level.

Reply:
Actually I agree with you. Thanks